### ISSUE TITLE
Bundler could not find compatible versions for gem "turbolinks-source"

### ISSUE THREAD

**Issue by speckins** [bundler/bundler#5132]
I'm encountering an issue while creating a new Rails 5 application on a Mac.  The "run bundle install" step fails to resolve dependencies.  However, "gem install" works.

```
$ chruby-exec ruby-2.3.1 -- rails new test-project
      create
         ...
         run  bundle install
Fetching gem metadata from https://rubygems.org/..........
Fetching version metadata from https://rubygems.org/..
Fetching dependency metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "turbolinks-source":
  In Gemfile:
    turbolinks (~> 5) was resolved to 5.0.0, which depends on
      turbolinks-source (~> 5)

Could not find gem 'turbolinks-source (~> 5)', which is required by gem 'turbolinks (~> 5)', in any of the sources.
```

Relevant line from Gemfile:
```
# Gemfile:
gem 'turbolinks', '~> 5'
```

macOS:  10.9.5
chruby:  0.3.9
ruby-install: 0.6.0
ruby: 2.3.1
rails:  5.0.0.1
bundler: 1.12.5, 1.13.6

I re-installed ruby-2.3.1, uninstalled all existing gems, re-ran "rails new", and the problem persists.

I tried replicating this issue in an Alpine Linux installation but was unsuccessful (Bundler is able to resolve the deps and "rails new" is successful).

I'm not sure if this is a Bundler issue or an error in turbolinks dependencies.  However, "gem install" doesn't have a problem resolving dependencies, so it seems like Bundler.  Version 5.0.0 of turbolink-source [exists on RubyGems](https://rubygems.org/gems/turbolinks-source/versions) and hasn't been yanked.

This works:
```
$ gem install -v5.0.0 turbolinks
Fetching: turbolinks-source-5.0.0.gem (100%)
Successfully installed turbolinks-source-5.0.0
Fetching: turbolinks-5.0.0.gem (100%)
Successfully installed turbolinks-5.0.0
```

So does this:
```
$ gem install -v5.0.1 turbolinks
Fetching: turbolinks-source-5.0.0.gem (100%)
Successfully installed turbolinks-source-5.0.0
Fetching: turbolinks-5.0.1.gem (100%)
Successfully installed turbolinks-5.0.1
2 gems installed
```

While I could get by with installing it manually on my development system, Capistrano deployment is failing with the same error.

**Comment by segiddins** [comment]
I was unable to reproduce this locally with the given Gemfile

**Comment by speckins** [comment]
I can reproduce this if I blow away my local Ruby and gems dirs and attempt to run Bundler on a Gemfile containing only "turbolinks."

```
$ rm -rf ~/.rubies/ruby-2.3.1/ ~/.gem/ruby/2.3.1
$ ruby-install ruby 2.3.1
# ...lots of time, lots of output...
$ mkdir ~/Projects/test-project
$ echo "ruby-2.3.1" > ~/Projects/test-project/.ruby-version
$ echo -e "source 'https://rubygems.org'\ngem 'turbolinks', '~> 5'" > ~/Projects/test-project/Gemfile
$ cd ~/Projects/test-project
$ chruby
   ruby-1.9.3-p545
   ruby-2.1.5
   ruby-2.2.1
   ruby-2.2.4
 * ruby-2.3.1
$ gem install bundler
Successfully installed bundler-1.13.6
1 gem installed
$ bundle install
Fetching gem metadata from https://rubygems.org/................
Fetching version metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "turbolinks-source":
  In Gemfile:
    turbolinks (~> 5) was resolved to 5.0.0, which depends on
      turbolinks-source (~> 5)

Could not find gem 'turbolinks-source (~> 5)', which is required by gem 'turbolinks (~> 5)', in any of the sources.
```

Apart from completely removing and reinstalling both the Ruby installation and the gems, is there anything else you can think of that might affect/cause this?  Anything else I can delete?

Like I said, I was able to set up a similar environment on an Alpine Linux installation in Docker, and that did not exhibit the same issue, so it might not be Bundler, but I'm at a loss as to what to try next.  Could it be a dependency misconfiguration in the "turbolinks" gem?

It appears I'm not the only one who has run into this recently:

- https://stackoverflow.com/questions/40084028/capistrano-could-not-find-turbolinks-source-5-0-0-in-any-of-the-sources
- https://stackoverflow.com/questions/40101183/could-not-find-gem-turbolinks-source-5

**Comment by segiddins** [comment]
Can you please share the output of `DEBUG_RESOLVER=1 bundle install --verbose` on the machine that's failing installation? Thanks!

**Comment by speckins** [comment]
Certainly.  This was run in a directory containing only .ruby-version and Gemfile, with Gemfile containing only the rubygems.org source and turbolinks gem (as in my previous comment).

```
$ DEBUG_RESOLVER=1 bundle install --verbose
Running `bundle install --verbose` with bundler 1.13.6
Found changes from the lockfile, re-resolving dependencies because the list of sources changed, the dependencies in your gemfile changed, you added a new platform to your gemfile
HTTP GET https://index.rubygems.org/versions
HTTP 206 Partial Content https://index.rubygems.org/versions
Fetching gem metadata from https://rubygems.org/
Looking up gems ["turbolinks"]
HTTP GET https://index.rubygems.org/info/turbolinks
HTTP 304 Not Modified https://index.rubygems.org/info/turbolinks
Looking up gems ["coffee-rails", "turbolinks-source"]
HTTP GET https://index.rubygems.org/info/coffee-rails
HTTP 304 Not Modified https://index.rubygems.org/info/coffee-rails
Looking up gems ["actionpack", "coffee-script", "railties", "sprockets"]
HTTP GET https://index.rubygems.org/info/coffee-script
HTTP 304 Not Modified https://index.rubygems.org/info/coffee-script
Looking up gems ["activesupport", "rake", "thor", "rack-ssl", "rdoc", "method_source", "hike", "rack", "tilt", "multi_json", "concurrent-ruby", "activemodel", "erubis", "rack-mount", "rack-test", "builder", "i18n", "tzinfo", "rack-cache", "journey", "actionview", "rails-deprecated_sanitizer", "rails-dom-testing", "rails-html-sanitizer", "coffee-script-source", "execjs"]
HTTP GET https://index.rubygems.org/info/rack-test
HTTP GET https://index.rubygems.org/info/multi_json
HTTP GET https://index.rubygems.org/info/hike
HTTP GET https://index.rubygems.org/info/rails-deprecated_sanitizer
HTTP GET https://index.rubygems.org/info/rails-dom-testing
HTTP GET https://index.rubygems.org/info/rails-html-sanitizer
HTTP GET https://index.rubygems.org/info/journey
HTTP GET https://index.rubygems.org/info/rack-ssl
HTTP GET https://index.rubygems.org/info/method_source
HTTP GET https://index.rubygems.org/info/builder
HTTP GET https://index.rubygems.org/info/erubis
HTTP GET https://index.rubygems.org/info/rack-cache
HTTP GET https://index.rubygems.org/info/tilt
HTTP GET https://index.rubygems.org/info/coffee-script-source
HTTP GET https://index.rubygems.org/info/thor
HTTP GET https://index.rubygems.org/info/rack-mount
HTTP GET https://index.rubygems.org/info/concurrent-ruby
HTTP GET https://index.rubygems.org/info/rack
HTTP GET https://index.rubygems.org/info/tzinfo
HTTP GET https://index.rubygems.org/info/actionview
HTTP GET https://index.rubygems.org/info/i18n
HTTP GET https://index.rubygems.org/info/execjs
HTTP 304 Not Modified https://index.rubygems.org/info/rails-deprecated_sanitizer
HTTP 304 Not Modified https://index.rubygems.org/info/tzinfo
HTTP 304 Not Modified https://index.rubygems.org/info/i18n
HTTP 304 Not Modified https://index.rubygems.org/info/rack-test
HTTP 304 Not Modified https://index.rubygems.org/info/hike
HTTP 304 Not Modified https://index.rubygems.org/info/multi_json
HTTP 304 Not Modified https://index.rubygems.org/info/journey
HTTP 304 Not Modified https://index.rubygems.org/info/rack-ssl
HTTP 304 Not Modified https://index.rubygems.org/info/concurrent-ruby
HTTP 304 Not Modified https://index.rubygems.org/info/rails-dom-testing
HTTP 304 Not Modified https://index.rubygems.org/info/tilt
HTTP 304 Not Modified https://index.rubygems.org/info/builder
HTTP 304 Not Modified https://index.rubygems.org/info/rails-html-sanitizer
HTTP 304 Not Modified https://index.rubygems.org/info/method_source
HTTP 304 Not Modified https://index.rubygems.org/info/thor
HTTP 304 Not Modified https://index.rubygems.org/info/rack-mount
HTTP 304 Not Modified https://index.rubygems.org/info/execjs
HTTP 304 Not Modified https://index.rubygems.org/info/actionview
HTTP 304 Not Modified https://index.rubygems.org/info/rack
HTTP 304 Not Modified https://index.rubygems.org/info/rack-cache
HTTP 304 Not Modified https://index.rubygems.org/info/erubis
HTTP 304 Not Modified https://index.rubygems.org/info/coffee-script-source
Looking up gems ["hoe", "minitest", "json", "bcrypt-ruby", "memcache-client", "thread_safe", "functional-ruby", "ref", "nokogiri", "loofah", "ruby_parser", "multimap", "camping", "fcgi", "mongrel", "ruby-openid", "test-spec", "thin", "abstract"]
HTTP GET https://index.rubygems.org/info/ref
HTTP GET https://index.rubygems.org/info/fcgi
HTTP GET https://index.rubygems.org/info/thread_safe
HTTP GET https://index.rubygems.org/info/camping
HTTP GET https://index.rubygems.org/info/loofah
HTTP GET https://index.rubygems.org/info/multimap
HTTP GET https://index.rubygems.org/info/functional-ruby
HTTP GET https://index.rubygems.org/info/memcache-client
HTTP GET https://index.rubygems.org/info/bcrypt-ruby
HTTP GET https://index.rubygems.org/info/json
HTTP GET https://index.rubygems.org/info/mongrel
HTTP GET https://index.rubygems.org/info/abstract
HTTP GET https://index.rubygems.org/info/ruby-openid
HTTP GET https://index.rubygems.org/info/test-spec
HTTP GET https://index.rubygems.org/info/thin
HTTP 304 Not Modified https://index.rubygems.org/info/fcgi
HTTP 304 Not Modified https://index.rubygems.org/info/camping
HTTP 304 Not Modified https://index.rubygems.org/info/ref
HTTP 304 Not Modified https://index.rubygems.org/info/loofah
HTTP 304 Not Modified https://index.rubygems.org/info/multimap
HTTP 304 Not Modified https://index.rubygems.org/info/bcrypt-ruby
HTTP 304 Not Modified https://index.rubygems.org/info/functional-ruby
HTTP 304 Not Modified https://index.rubygems.org/info/test-spec
HTTP 304 Not Modified https://index.rubygems.org/info/thin
HTTP 304 Not Modified https://index.rubygems.org/info/memcache-client
HTTP 304 Not Modified https://index.rubygems.org/info/abstract
HTTP 304 Not Modified https://index.rubygems.org/info/ruby-openid
HTTP 304 Not Modified https://index.rubygems.org/info/thread_safe
HTTP 304 Not Modified https://index.rubygems.org/info/mongrel
HTTP 304 Not Modified https://index.rubygems.org/info/json
Looking up gems ["ParseTree", "sexp_processor", "racc", "rake-compiler", "rexical", "tenderlove-frex", "weakling", "mini_portile", "mini_portile2", "pkg-config", "rubyforge", "RubyInline", "gemcutter", "markaby", "metaid", "activerecord", "mab", "crass", "flexmock", "mocha", "bcrypt", "daemons", "eventmachine", "eventmachine-le", "http_parser.rb", "preforker", "ZenTest", "ruby-yadis", "atomic", "cgi_multipart_eof_fix", "fastthread", "gem_plugin"]
HTTP GET https://index.rubygems.org/info/rexical
HTTP GET https://index.rubygems.org/info/racc
HTTP GET https://index.rubygems.org/info/gemcutter
HTTP GET https://index.rubygems.org/info/pkg-config
HTTP GET https://index.rubygems.org/info/tenderlove-frex
HTTP GET https://index.rubygems.org/info/mini_portile2
HTTP GET https://index.rubygems.org/info/sexp_processor
HTTP GET https://index.rubygems.org/info/ParseTree
HTTP GET https://index.rubygems.org/info/rubyforge
HTTP GET https://index.rubygems.org/info/crass
HTTP GET https://index.rubygems.org/info/bcrypt
HTTP GET https://index.rubygems.org/info/RubyInline
HTTP GET https://index.rubygems.org/info/rake-compiler
HTTP GET https://index.rubygems.org/info/markaby
HTTP GET https://index.rubygems.org/info/metaid
HTTP GET https://index.rubygems.org/info/daemons
HTTP GET https://index.rubygems.org/info/weakling
HTTP GET https://index.rubygems.org/info/eventmachine-le
HTTP GET https://index.rubygems.org/info/http_parser.rb
HTTP GET https://index.rubygems.org/info/mab
HTTP GET https://index.rubygems.org/info/eventmachine
HTTP GET https://index.rubygems.org/info/ZenTest
HTTP GET https://index.rubygems.org/info/preforker
HTTP GET https://index.rubygems.org/info/atomic
HTTP GET https://index.rubygems.org/info/ruby-yadis
HTTP 304 Not Modified https://index.rubygems.org/info/tenderlove-frex
HTTP 304 Not Modified https://index.rubygems.org/info/gemcutter
HTTP 304 Not Modified https://index.rubygems.org/info/racc
HTTP 304 Not Modified https://index.rubygems.org/info/bcrypt
HTTP 304 Not Modified https://index.rubygems.org/info/mini_portile2
HTTP 304 Not Modified https://index.rubygems.org/info/sexp_processor
HTTP 304 Not Modified https://index.rubygems.org/info/ParseTree
HTTP 304 Not Modified https://index.rubygems.org/info/rubyforge
HTTP 304 Not Modified https://index.rubygems.org/info/rake-compiler
HTTP 304 Not Modified https://index.rubygems.org/info/RubyInline
HTTP 304 Not Modified https://index.rubygems.org/info/crass
HTTP 304 Not Modified https://index.rubygems.org/info/markaby
HTTP 304 Not Modified https://index.rubygems.org/info/metaid
HTTP 304 Not Modified https://index.rubygems.org/info/weakling
HTTP 304 Not Modified https://index.rubygems.org/info/daemons
HTTP 304 Not Modified https://index.rubygems.org/info/eventmachine-le
HTTP 304 Not Modified https://index.rubygems.org/info/pkg-config
HTTP 304 Not Modified https://index.rubygems.org/info/ZenTest
HTTP GET https://index.rubygems.org/info/fastthread
HTTP 304 Not Modified https://index.rubygems.org/info/mab
HTTP GET https://index.rubygems.org/info/cgi_multipart_eof_fix
HTTP 304 Not Modified https://index.rubygems.org/info/http_parser.rb
HTTP 304 Not Modified https://index.rubygems.org/info/ruby-yadis
HTTP 304 Not Modified https://index.rubygems.org/info/atomic
HTTP GET https://index.rubygems.org/info/gem_plugin
HTTP 304 Not Modified https://index.rubygems.org/info/eventmachine
HTTP 304 Not Modified https://index.rubygems.org/info/preforker
HTTP 304 Not Modified https://index.rubygems.org/info/fastthread
HTTP 304 Not Modified https://index.rubygems.org/info/gem_plugin
HTTP 304 Not Modified https://index.rubygems.org/info/cgi_multipart_eof_fix
HTTP 304 Not Modified https://index.rubygems.org/info/rexical
Looking up gems ["arel", "activerecord-deprecated_finders", "metaclass", "net-scp", "json_pure", "SexpProcessor"]
HTTP GET https://index.rubygems.org/info/activerecord-deprecated_finders
HTTP GET https://index.rubygems.org/info/json_pure
HTTP GET https://index.rubygems.org/info/SexpProcessor
HTTP GET https://index.rubygems.org/info/metaclass
HTTP GET https://index.rubygems.org/info/net-scp
HTTP 304 Not Modified https://index.rubygems.org/info/activerecord-deprecated_finders
HTTP 304 Not Modified https://index.rubygems.org/info/json_pure
HTTP 304 Not Modified https://index.rubygems.org/info/net-scp
HTTP 304 Not Modified https://index.rubygems.org/info/metaclass
HTTP 304 Not Modified https://index.rubygems.org/info/SexpProcessor
Looking up gems ["spruz", "net-ssh"]
HTTP GET https://index.rubygems.org/info/spruz
HTTP GET https://index.rubygems.org/info/net-ssh
HTTP 304 Not Modified https://index.rubygems.org/info/spruz
HTTP 304 Not Modified https://index.rubygems.org/info/net-ssh
Looking up gems ["echoe", "needle", "jruby-pageant", "bcrypt_pbkdf", "rbnacl", "rbnacl-libsodium"]
HTTP GET https://index.rubygems.org/info/jruby-pageant
HTTP GET https://index.rubygems.org/info/needle
HTTP GET https://index.rubygems.org/info/echoe
HTTP GET https://index.rubygems.org/info/rbnacl-libsodium
HTTP GET https://index.rubygems.org/info/rbnacl
HTTP 304 Not Modified https://index.rubygems.org/info/needle
HTTP 304 Not Modified https://index.rubygems.org/info/jruby-pageant
HTTP 304 Not Modified https://index.rubygems.org/info/rbnacl-libsodium
HTTP 304 Not Modified https://index.rubygems.org/info/echoe
HTTP 304 Not Modified https://index.rubygems.org/info/rbnacl
Looking up gems ["highline", "rcov", "allison", "ffi"]
HTTP GET https://index.rubygems.org/info/rcov
HTTP GET https://index.rubygems.org/info/allison
HTTP GET https://index.rubygems.org/info/highline
HTTP GET https://index.rubygems.org/info/ffi
HTTP 304 Not Modified https://index.rubygems.org/info/ffi
HTTP 304 Not Modified https://index.rubygems.org/info/allison
HTTP 304 Not Modified https://index.rubygems.org/info/highline
HTTP 304 Not Modified https://index.rubygems.org/info/rcov
Looking up gems ["bones", "termios"]
HTTP GET https://index.rubygems.org/info/termios
HTTP GET https://index.rubygems.org/info/bones
HTTP 304 Not Modified https://index.rubygems.org/info/termios
HTTP 304 Not Modified https://index.rubygems.org/info/bones
Looking up gems ["little-plugger", "loquacious", "bones-extras", "bones-git", "rspec"]
HTTP GET https://index.rubygems.org/info/bones-extras
HTTP GET https://index.rubygems.org/info/little-plugger
HTTP GET https://index.rubygems.org/info/loquacious
HTTP GET https://index.rubygems.org/info/bones-git
HTTP 304 Not Modified https://index.rubygems.org/info/little-plugger
HTTP 304 Not Modified https://index.rubygems.org/info/loquacious
HTTP 304 Not Modified https://index.rubygems.org/info/bones-git
HTTP 304 Not Modified https://index.rubygems.org/info/bones-extras
Looking up gems ["cucumber", "diff-lcs", "spicycode-rcov", "syntax", "rspec-core", "rspec-expectations", "rspec-mocks", "git", "bones-rcov", "bones-rspec", "bones-rubyforge", "bones-zentest"]
HTTP GET https://index.rubygems.org/info/syntax
HTTP GET https://index.rubygems.org/info/bones-zentest
HTTP GET https://index.rubygems.org/info/git
HTTP GET https://index.rubygems.org/info/spicycode-rcov
HTTP GET https://index.rubygems.org/info/bones-rspec
HTTP GET https://index.rubygems.org/info/bones-rubyforge
HTTP GET https://index.rubygems.org/info/bones-rcov
HTTP GET https://index.rubygems.org/info/diff-lcs
HTTP 304 Not Modified https://index.rubygems.org/info/syntax
HTTP 304 Not Modified https://index.rubygems.org/info/bones-zentest
HTTP 304 Not Modified https://index.rubygems.org/info/git
HTTP 304 Not Modified https://index.rubygems.org/info/spicycode-rcov
HTTP 304 Not Modified https://index.rubygems.org/info/bones-rubyforge
HTTP 304 Not Modified https://index.rubygems.org/info/bones-rcov
HTTP 304 Not Modified https://index.rubygems.org/info/bones-rspec
HTTP 304 Not Modified https://index.rubygems.org/info/diff-lcs
Looking up gems ["rspec-support", "polyglot", "term-ansicolor", "treetop", "gherkin", "multi_test", "cucumber-core", "gherkin3", "cucumber-wire", "event-bus"]
HTTP GET https://index.rubygems.org/info/event-bus
HTTP GET https://index.rubygems.org/info/treetop
HTTP GET https://index.rubygems.org/info/cucumber-wire
HTTP GET https://index.rubygems.org/info/multi_test
HTTP GET https://index.rubygems.org/info/polyglot
HTTP GET https://index.rubygems.org/info/gherkin3
HTTP GET https://index.rubygems.org/info/gherkin
HTTP 304 Not Modified https://index.rubygems.org/info/cucumber-wire
HTTP 304 Not Modified https://index.rubygems.org/info/treetop
HTTP 304 Not Modified https://index.rubygems.org/info/multi_test
HTTP 304 Not Modified https://index.rubygems.org/info/polyglot
HTTP 304 Not Modified https://index.rubygems.org/info/event-bus
HTTP 304 Not Modified https://index.rubygems.org/info/gherkin3
HTTP 304 Not Modified https://index.rubygems.org/info/gherkin
Looking up gems ["backports", "tins", "facets", "trollop"]
HTTP GET https://index.rubygems.org/info/facets
HTTP GET https://index.rubygems.org/info/tins
HTTP GET https://index.rubygems.org/info/backports
HTTP GET https://index.rubygems.org/info/trollop
HTTP 304 Not Modified https://index.rubygems.org/info/backports
HTTP 304 Not Modified https://index.rubygems.org/info/trollop
HTTP 304 Not Modified https://index.rubygems.org/info/facets
HTTP 304 Not Modified https://index.rubygems.org/info/tins
Looking up gems ["curses"]
HTTP GET https://index.rubygems.org/info/curses
HTTP 304 Not Modified https://index.rubygems.org/info/curses
Fetching version metadata from https://rubygems.org/
Looking up gems ["coffee-rails", "turbolinks-source", "actionpack", "coffee-script", "railties", "sprockets", "activesupport", "rake", "thor", "rack-ssl", "rdoc", "method_source", "hike", "rack", "tilt", "multi_json", "concurrent-ruby", "activemodel", "erubis", "rack-mount", "rack-test", "builder", "i18n", "tzinfo", "rack-cache", "journey", "actionview", "rails-deprecated_sanitizer", "rails-dom-testing", "rails-html-sanitizer", "coffee-script-source", "execjs", "hoe", "minitest", "json", "bcrypt-ruby", "memcache-client", "thread_safe", "functional-ruby", "ref", "nokogiri", "loofah", "ruby_parser", "multimap", "camping", "fcgi", "mongrel", "ruby-openid", "test-spec", "thin", "abstract", "ParseTree", "sexp_processor", "racc", "rake-compiler", "rexical", "tenderlove-frex", "weakling", "mini_portile", "mini_portile2", "pkg-config", "rubyforge", "RubyInline", "gemcutter", "markaby", "metaid", "activerecord", "mab", "crass", "flexmock", "mocha", "bcrypt", "daemons", "eventmachine", "eventmachine-le", "http_parser.rb", "preforker", "ZenTest", "ruby-yadis", "atomic", "cgi_multipart_eof_fix", "fastthread", "gem_plugin", "arel", "activerecord-deprecated_finders", "metaclass", "net-scp", "json_pure", "SexpProcessor", "spruz", "net-ssh", "echoe", "needle", "jruby-pageant", "bcrypt_pbkdf", "rbnacl", "rbnacl-libsodium", "highline", "rcov", "allison", "ffi", "bones", "termios", "little-plugger", "loquacious", "bones-extras", "bones-git", "rspec", "cucumber", "diff-lcs", "spicycode-rcov", "syntax", "rspec-core", "rspec-expectations", "rspec-mocks", "git", "bones-rcov", "bones-rspec", "bones-rubyforge", "bones-zentest", "rspec-support", "polyglot", "term-ansicolor", "treetop", "gherkin", "multi_test", "cucumber-core", "gherkin3", "cucumber-wire", "event-bus", "backports", "tins", "facets", "trollop", "curses"]
Starting resolution (2016-10-27 18:54:12 -0500)
Resolving dependencies...Creating possibility state for turbolinks (~> 5) (2 remaining)
  Attempting to activate turbolinks (5.0.1)
  Activated turbolinks at turbolinks (5.0.1)
  Requiring nested dependencies (turbolinks-source (~> 5))
  Creating possibility state for turbolinks-source (~> 5) (0 remaining)
    Unwinding for conflict: turbolinks-source (~> 5)
Creating possibility state for turbolinks (~> 5) (1 remaining)
  Attempting to activate turbolinks (5.0.0)
  Activated turbolinks at turbolinks (5.0.0)
  Requiring nested dependencies (turbolinks-source (~> 5))
  Creating possibility state for turbolinks-source (~> 5) (0 remaining)
    Unwinding for conflict: turbolinks-source (~> 5)

Finished resolution (4 steps) (Took 0.001791 seconds) (2016-10-27 18:54:12 -0500)
Bundler could not find compatible versions for gem "turbolinks-source":
  In Gemfile:
    turbolinks (~> 5) was resolved to 5.0.0, which depends on
      turbolinks-source (~> 5)

Could not find gem 'turbolinks-source (~> 5)', which is required by gem 'turbolinks (~> 5)', in any of the sources.
```

For fun, I created a gist of the above result and a passing result from the Alpine container:  https://gist.github.com/speckins/3e4daf3dc12f2dfc9573624267860ab7/revisions

**Comment by segiddins** [comment]
Just for fun, since I can't reproduce the issue, what happens if you only depend upon `turbolinks-source (~> 5)` in your gemfile?

**Comment by speckins** [comment]
**Gemfile**
```
source 'https://rubygems.org'
gem 'turbolinks-source', '~> 5'
```

**Output (Mac)**
```
$ DEBUG_RESOLVER=1 bundle install --verbose
Running `bundle install --verbose` with bundler 1.13.6
Found changes from the lockfile, re-resolving dependencies because the list of sources changed, the dependencies in your gemfile changed, you added a new platform to your gemfile
HTTP GET https://index.rubygems.org/versions
HTTP 206 Partial Content https://index.rubygems.org/versions
Fetching gem metadata from https://rubygems.org/
Looking up gems ["turbolinks-source"]
Fetching version metadata from https://rubygems.org/
Could not find gem 'turbolinks-source (~> 5)' in any of the gem sources listed in your Gemfile or available on this machine.

$ DEBUG_RESOLVER=1 bundle install --verbose
Running `bundle install --verbose` with bundler 1.13.6
Found changes from the lockfile, re-resolving dependencies because the list of sources changed, the dependencies in your gemfile changed, you added a new platform to your gemfile
HTTP GET https://index.rubygems.org/versions
HTTP 304 Not Modified https://index.rubygems.org/versions
Fetching gem metadata from https://rubygems.org/
Looking up gems ["turbolinks-source"]
Fetching version metadata from https://rubygems.org/
Could not find gem 'turbolinks-source (~> 5)' in any of the gem sources listed in your Gemfile or available on this machine.
```

**Output (Linux)**
```
$ DEBUG_RESOLVER=1 bundle install --verbose
Running `bundle install --verbose` with bundler 1.13.6
Found changes from the lockfile, re-resolving dependencies because the list of sources changed, the dependencies in your gemfile changed, you added a new platform to your gemfile
HTTP GET https://index.rubygems.org/versions
HTTP 304 Not Modified https://index.rubygems.org/versions
Fetching gem metadata from https://rubygems.org/
Looking up gems ["turbolinks-source"]
HTTP GET https://index.rubygems.org/info/turbolinks-source
HTTP 304 Not Modified https://index.rubygems.org/info/turbolinks-source
Fetching version metadata from https://rubygems.org/
Starting resolution (2016-10-28 01:18:03 +0000)
Resolving dependencies...Creating possibility state for turbolinks-source (~> 5) (1 remaining)
  Attempting to activate turbolinks-source (5.0.0)
  Activated turbolinks-source at turbolinks-source (5.0.0)
  Requiring nested dependencies ()

Finished resolution (1 steps) (Took 0.00548569 seconds) (2016-10-28 01:18:03 +0000)
Unactivated:
Activated: turbolinks-source
Using turbolinks-source 5.0.0
0:  turbolinks-source (5.0.0) from /home/me/.gem/ruby/2.3.1/specifications/turbolinks-source-5.0.0.gemspec
Using bundler 1.13.6
0:  bundler (1.13.6) from /home/me/.gem/ruby/2.3.1/specifications/bundler-1.13.6.gemspec
Bundle complete! 1 Gemfile dependency, 2 gems now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.
```

**Comment by speckins** [comment]
This morning it works.

```
$ DEBUG_RESOLVER=1 bundle install --verbose
Running `bundle install --verbose` with bundler 1.13.6
Found changes from the lockfile, re-resolving dependencies because the list of sources changed, the dependencies in your gemfile changed, you added a new platform to your gemfile
HTTP GET https://index.rubygems.org/versions
HTTP 206 Partial Content https://index.rubygems.org/versions
Fetching gem metadata from https://rubygems.org/
Looking up gems ["turbolinks-source"]
Fetching version metadata from https://rubygems.org/
Starting resolution (2016-10-28 09:58:30 -0500)
Resolving dependencies...Creating possibility state for turbolinks-source (~> 5) (1 remaining)
  Attempting to activate turbolinks-source (5.0.0)
  Activated turbolinks-source at turbolinks-source (5.0.0)
  Requiring nested dependencies ()

Finished resolution (1 steps) (Took 0.00129 seconds) (2016-10-28 09:58:30 -0500)
Unactivated:
Activated: turbolinks-source
Using turbolinks-source 5.0.0
0:  turbolinks-source (5.0.0) from /Users/speckins/.gem/ruby/2.3.1/specifications/turbolinks-source-5.0.0.gemspec
Using bundler 1.13.6
0:  bundler (1.13.6) from /Users/speckins/.gem/ruby/2.3.1/specifications/bundler-1.13.6.gemspec
Bundle complete! 1 Gemfile dependency, 2 gems now installed.
Use `bundle show [gemname]` to see where a bundled gem is installed.
```

**Comment by speckins** [comment]
Okay, I think I got this.

I don't know why it worked that one time (previous comment).  I uninstalled and reinstalled Ruby again, and it was back to failing.

Figuring it was something in my $HOME or environment, I created a new user on my Mac, installed Ruby, and it worked.  That led me to realize that bundler keeps its cache in ~/.bundle/cache, so I removed that from my normal account, and now it works as expected.

I'm a little surprised this affected a "bundle install --path" command; I thought that was more contained.  (I was testing Capistrano by installing to localhost.)
